fluentd.gemspec

```diff
@@ -29,6 +29,9 @@ Gem::Specification.new do |gem|
   gem.add_runtime_dependency("tzinfo-data", ["~> 1.0"])
   gem.add_runtime_dependency("strptime", [">= 0.2.4", "< 1.0.0"])
   gem.add_runtime_dependency("webrick", ["~> 1.4"])
+  gem.add_runtime_dependency("aws-sigv4", ["~> 1.8"])
+  gem.add_runtime_dependency("aws-sdk-sts", ["~> 1.11"])
+  gem.add_runtime_dependency("rexml", ["~> 3.2"])
```

Comment:
We want to minimize mandatory requirements for installing fluentd core.
So it would be better that make these gems optional.
For example:

* Remove these gems from fluentd.gemspec
* Do `require 'aws-sigv4'` and `require 'aws-sdk-core'` only when AWS related parameters are set.
* Show error log then prompt users to install these gems when `require` is failed.

Reply:
@ashie , Thank you for the solid suggestions. I've made these changes accordingly. The one difference is that I moved these dependencies to development dependencies so that the tests run.

Reply:
It seems nice. Thanks!